### PR TITLE
Fix missing NONC tag in responses and wrong type

### PR DESCRIPTION
**Pull request by dansarie**
This PR fixes two bugs found during the IETF Hackathon in Dublin.
* Responses MUST contain the NONC tag, see Section [6.2.3.](https://datatracker.ietf.org/doc/html/draft-ietf-ntp-roughtime-11#section-6.2.3) of draft-11.
* The type of the srv struct member is of the wrong type - it should be a byte array.

### CODE REVIEW

File: protocol/protocol.go [hunk 1]
```diff
@@ -625,6 +625,8 @@ func CreateReplies(ver Version, requests []Request, midpoint time.Time, radius t
 		binary.LittleEndian.PutUint32(indexBytes[:], uint32(i))
 		reply[tagINDX] = indexBytes[:]
 
+		reply[tagNONC] = requests[i].Nonce
```
Comment (lukevalenta) [inline protocol/protocol.go:628]: Thanks! Looks like the Google-Roughtime it's not required to send `NONC` back, but also not forbidden: https://roughtime.googlesource.com/roughtime/+/refs/heads/master/PROTOCOL.md#responses, so this seems OK.

We claim at https://github.com/cloudflare/roughtime/blob/master/protocol/protocol.go#L811 that `VerifyReply` checks that the server response includes a nonce, but it doesn't look like we actually perform that check (which is why the current client/server implementations in this repo interoperate).

@dansarie would you want to add that check in to this PR, perhaps at https://github.com/cloudflare/roughtime/blob/master/protocol/protocol.go#L969?

Comment (dansarie) [reply protocol/protocol.go:628]: @lukevalenta Okay! Should be fixed now.

**Review by lukevalenta — APPROVED** [review]
Looks good to me--thank you!

**Comment by lukevalenta** [comment]
Some of the CI tests are failing--maybe test vectors need to be updated. I'll investigate soon.

File: protocol/protocol.go [hunk 2]
```diff
@@ -853,6 +855,17 @@ func VerifyReply(versionPreference []Version, replyBytes, publicKey []byte, nonc
 		return midp, radi, errUnsupportedVersion([]Version{responseVer})
 	}
 
+	// Make sure the NONC tag is present and indicates the same nonce as sent in request
+	if versionIETF {
+		responseNonc, ok := reply[tagNONC]
+		if !ok {
+			return midp, radi, errors.New("protocol: response is missing NONC tag")
+		}
+		if !bytes.Equal(nonce, responseNonc) {
```
Comment (cjpatton) [inline protocol/protocol.go:864]: nit: "Nonc" should be capitalized to match `tagNONC`. Or go with `responseNonse`?

**Review by cjpatton — COMMENTED** [review]
Tests should pass after regenerating the test vectors:
```bash
cd protocol && go generate
```

File: protocol/protocol.go (hunk 2 above)
Comment (dansarie) [reply protocol/protocol.go]: Fixed by 96d8bf00b783f1edc66ea04ceba5dd12bc928de2.

File: protocol/protocol.go [hunk 3]
```diff
@@ -853,6 +855,17 @@ func VerifyReply(versionPreference []Version, replyBytes, publicKey []byte, nonc
 		return midp, radi, errUnsupportedVersion([]Version{responseVer})
 	}
 
+	// Make sure the NONC tag is present and indicates the same nonce as sent in request
```
[the remark this answers is not in the document]
Comment (dansarie) [reply protocol/protocol.go]: Fixed by 96d8bf00b783f1edc66ea04ceba5dd12bc928de2.

**Comment by dansarie** [comment]
Nits fixed and test vectors updated!

**Pull request merged**